backend/api/observability.py

```
@@ -0,0 +1,229 @@
+
+
+def _endpoint_host(endpoint: str | None) -> str | None:
+    if not endpoint or not endpoint.strip():
+        return None
+    parsed = urlparse(endpoint.strip())
+    if not parsed.netloc:
+        return None
+    return parsed.netloc.rsplit("@", 1)[-1]
+
+
+def _check_org_admin(auth_context: AuthContext = Depends(get_auth_context)) -> AuthContext:
```

Ruff (0.15.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)